### DISCUSSION TITLE
How to log requests

### DISCUSSION THREAD

**Discussion by lokeshwarobuli** [PyGithub/PyGithub#2737] [Q&A]
I just wanna log only the requests out of pygithub and I have used `enable_console_debug_logging()` but it logs entire thing which I don't want. Any Ideas. Thanks inadvance

**Comment by EnricoMi** [comment]
You could define your own log function:

```python
def log(verb, url, requestHeaders, input, status, responseHeaders, output):
    print(f"verb={verb}, url={url}, status={status}")
```

And then patch the requester's log method:
```python
g._Github__requester._Requester__log = log
```

You then have full control over what and how to log each request:
```python
g.get_repo("PyGithub/Pygithub").owner
```
```
verb=GET, url=/repos/PyGithub/Pygithub, status=200
```

**Reply by lokeshwarobuli** [reply]
Thanks a lot, will try this out and reach back incase I am stuck !!
